fuzzy-native/src/MatcherBase.cpp

```diff
@@ -35,12 +35,51 @@ inline string str_to_lower(const std::string &s) {
   return lower;
 }
 
+float distance_to_root(const std::string *root, const std::string *value) {
```

Comment:
Function parameters should be references (`const std::string &`) rather than pointers when possible (i.e. when they're non-NULL).